source/particle/output/ascii.cc

```diff
@@ -106,19 +99,15 @@ namespace aspect
             output << "\n";
           }
 
-        file_index++;
+        this->increment_file_index ();
 
-        return output_file_prefix;
+        return "txt";
       }
 
       template <int dim>
       template <class Archive>
-      void ASCIIOutput<dim>::serialize (Archive &ar, const unsigned int)
-      {
-        // invoke serialization of the base class
-        ar &file_index
-        ;
-      }
+      void ASCIIOutput<dim>::serialize (Archive &, const unsigned int)
+      {}
```

Comment:
I have moved the the function serialize into the base class, interface, and declared it as a virtual. Serializing the member variables, i.e:

```
      template <int dim>
      template <class Archive>
      void Interface<dim>::serialize (Archive &ar, const unsigned int)
      {
        ar &output_file_suffix
        &file_index
        ;
      }
```

In terms of serialization, is a top down model in that the parent classes serialize function is called followed by the serialize function of the childs class?